docs/content/direct/architecture.md

```diff
@@ -263,166 +327,98 @@ handler and the work queue as follows:
-       A delete event for that object consumed by the event handler enqueues 
-       a key for [Object Deleted](#object-deleted).
-    -  Gets the lister for BindingPolicy objects and list all binding-policies
-    -  Iterates on all binding-policies, and for each of them:
-        - Evaluates whether the object matches the downsync selection 
-          criteria in the `BindingPolicy`.
-        - Whether the object is a match or not, the worker notes it through the in-memory representation 
-       of the relevant `Binding`. If the noting of an object results in a change in the in-memory representation of the
-         `Binding`, the worker enqueues the latter for syncing.
-        - If a matched `BindingPolicy` has `WantSingletonReportedState` set to true (**see note below), the object is 
-       labeled with a special label in order to be able to track it for status reporting.
-        - If no matching `BindingPolicy` has `WantSingletonReportedState` set to true, the worker sets the label value
-       to false if the label exists.
-    - Worker returns and is ready to pick other keys from the queue.
-
-`WantSingletonReportedState`:
-currently it is the user's responsibility to make sure that a binding-policy that sets `WantSingletonReportedState` to
-true is not in conflict with other binding-policies that do the same, and that the binding-policy selects only a single
-cluster.
-
-There are other event flows, based on the object `GroupVersionKind` and type of event. 
-Error conditions may cause the re-enqueuing of keys, resulting in retries.
-The following sections broadly describe these flows.
-
-#### Object Deleted 
+The informer and watches specific resources on the WDS API Server; on
+create/update/delete object events it puts a copy of the object into
+the informer's local cache, which is what the lister reads. The
+informer invokes the event handler. The handler implements the event
+handling functions (`AddFunc`, `UpdateFunc`, `DeleteFunc`)
 
-When an object is deleted from the WDS, the handler’s `DeleteFunc` is
-invoked. A shallow copy of the object is added to a field in the key 
-before pushing it to the work queue. Then:
+#### Sync BindingPolicy
 
-- Worker pulls the key from the work queue
+For a `BindingPolicy` that is deleted or being deleted, sync consists of the following steps.
 
-- Flow continues the same way as in the create/update scenario, however
-  the `deletedObject` field in the key indicates that the object has been
-  deleted and that it needs to be removed from all clusters.
+1. Ensure the absence of the KubeStellar finalizer on the `BindingPolicy`.
+1. Invoke the `BindingPolicyResover` method for the absence of the `BindingPolicy`.
 
-- In-memory representations of affected bindings are updated to remove the object.
-If any are indeed affected, they are enqueued for syncing.
+For a `BindingPolicy` that is not deleted or being deleted, sync consists of the following steps.
```

Comment:
For accuracy this should say "is not deleted and is not being deleted".

Reply:
Revised.